src/apps/containers.clj

```diff
@@ -585,34 +599,210 @@
+
+(defn container-proxy-settings-uuid
+  "Returns the interapps-proxy-settings uuid for the given container-settings
+   UUID."
+  [container-settings-uuid]
+  (:interactive_apps_proxy_settings_id
+   (first (select container-settings
+                  (where {:id (uuidify container-settings-uuid)})))))
+
+(defn modify-proxy-settings
+  "Modifies an existing set of proxy settings. Needs the proxy-settings-uuid
+   and a new map of values for the proxy settings."
+  [proxy-settings-uuid proxy-settings]
+  (if-not (proxy-settings? proxy-settings-uuid)
+    (throw (Exception. (str "proxy settings not found:" proxy-settings-uuid))))
+  (sql/update interapps-proxy-settings
+              (set-fields (filter-proxy-settings proxy-settings))
+              (where {:id (uuidify proxy-settings-uuid)})))
+
+(defn container-proxy-settings
+  "Returns the proxy-settings for the given container-settings UUID."
+  [container-settings-uuid]
+  (first (select container-settings
+                 (with interapps-proxy-settings)
+                 (fields :interactive_apps_proxy_settings.id
+                         :interactive_apps_proxy_settings.image
+                         :interactive_apps_proxy_settings.name
+                         :interactive_apps_proxy_settings.frontend_url
+                         :interactive_apps_proxy_settings.cas_url
+                         :interactive_apps_proxy_settings.cas_validate
+                         :interactive_apps_proxy_settings.ssl_cert_path
+                         :interactive_apps_proxy_settings.ssl_key_path)
+                 (where {:id (uuidify container-settings-uuid)}))))
+
 (defn add-tool-container
   [tool-uuid info-map]
   (when (tool-has-settings? tool-uuid)
     (throw (Exception. (str "Tool " tool-uuid " already has container settings."))))
-  (let [devices  (:container_devices info-map)
-        volumes  (:container_volumes info-map)
-        vfs      (:container_volumes_from info-map)
-        settings (dissoc info-map :container_devices :container_volumes :container_volumes_from)
-        info-map (assoc info-map :tools_id (uuidify tool-uuid))]
+  (let [devices        (:container_devices info-map)
+        volumes        (:container_volumes info-map)
+        vfs            (:container_volumes_from info-map)
+        ports          (:container_ports info-map)
+        proxy-settings (:interactive_apps info-map)
+        settings       (dissoc info-map :container_devices :container_volumes :container_volumes_from)
+        info-map       (assoc info-map :tools_id (uuidify tool-uuid))]
     (log/warn "adding container information for tool" tool-uuid ":" info-map)
     (transaction
      (let [img-id        (find-or-add-image-id (:image info-map))
            settings-map  (add-settings info-map)
            settings-uuid (:id settings-map)]
        (update-tool {:id tool-uuid :container_images_id img-id})
+       (if-not (nil? proxy-settings)
```

Comment:
I like to use `when-not` instead of `if-not` when there's no `else` clause. When I see `if` or `if-not` being used, I tend to look for the else clause even if it's not there.